keyboards/jlw/vault35_katana/keymaps/vial/keymap.c

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 sporkus
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "../../rgb.c"
+
+#define HM_A	LSFT_T(KC_A)
+#define HM_S	LALT_T(KC_S)
+#define HM_D	LGUI_T(KC_D)
+#define HM_F	LCTL_T(KC_F)
+#define HM_J	RCTL_T(KC_J)
+#define HM_K	RGUI_T(KC_K)
+#define HM_L	RALT_T(KC_L)
+#define HM_Z	LSFT_T(KC_Z)
+#define HM_SLSH	LSFT_T(KC_SLSH)
+#define HM_QUOT	RSFT_T(KC_QUOT)
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_split_6bars(
+         KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,                   KC_Y,   KC_U,   KC_I,    KC_O,   KC_P,
+         KC_A,   KC_S,   KC_D,   KC_F,   KC_G,                   KC_H,   KC_J,   KC_K,    KC_L,   KC_QUOT,
+         KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,     RGB_TOG,      KC_N,   KC_M,   KC_COMM, KC_DOT, KC_SLSH,
+                 KC_ESC, MO(1),  KC_ENT,                         KC_SPC, MO(2),  KC_BSPC
+    ),
+
+    [1] = LAYOUT_split_6bars(
+         _______,  _______,  _______,  _______,  _______,            _______,  KC_7,     KC_8,     KC_9,     KC_0,
+         _______,  _______,  _______,  _______,  _______,            _______,  KC_4,     KC_5,     KC_6,     _______,
+         _______,  _______,  _______,  _______,  _______,   QK_BOOT, _______,  KC_1,     KC_2,     KC_3,     _______,
+                                       _______,  _______,            _______,  _______, _______,  _______
+    ),
+
+    [2] = LAYOUT_split_6bars(
+         _______,  _______,  _______,  _______,  _______,            _______,  _______,   _______,   _______,    _______,
+         _______,  _______,  _______,  _______,  _______,            KC_LEFT,  KC_DOWN,   KC_UP,     KC_LEFT,    _______,
+         _______,  _______,  _______,  _______,  _______,   QK_BOOT, _______,  _______,   _______,   _______,    _______,
+                                       _______,  _______,            _______,  _______, _______,  _______
+    )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+#undef BLANK_LAYER
+#define BLANK_LAYER { ENCODER_CCW_CW(_______, _______) }
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] =   BLANK_LAYER,
+    [2] =   BLANK_LAYER
```

Comment:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] =   { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
    [1] =   { ENCODER_CCW_CW(_______, _______) },
    [2] =   { ENCODER_CCW_CW(_______, _______) }
```
There really isn't any need to overcomplicate this.